### PR TITLE
Dropping Axelrod as an dependency 

### PR DESCRIPTION
**Pull request by Nikoleta-v3**

### CODE REVIEW

File: src/mockup.py [hunk 1]
```diff
@@ -1,14 +1,26 @@
 # %%
 import itertools
 
-import axelrod as axl
 import numpy as np
 #%%
-from axelrod.action import Action
+from enum import Enum
+# from axelrod.action import Action
```
Comment (Nikoleta-v3) [inline src/mockup.py]:
```suggestion
```

File: src/mockup.py [hunk 2]
```diff
@@ -182,40 +193,92 @@ def long_term_total_payoff(opening_payoffs, exp_p, delta):
     return payoffs + exp_p * delta ** len(opening_payoffs) / (1.0-delta)
 
 # %%
+def make_history(player1_moves, player2_mem1_strategy):
+    """
+    play the game between two players for `len(player1_moves)` turns and return the history
```
Comment (Nikoleta-v3) [inline src/mockup.py]:
```suggestion
    Plays the repeated game between a sequence player and a player with a memory-one strategy and returns the history. The repeated game is played for `len(player1_moves)` turns.
```

File: src/mockup.py [hunk 3]
```diff
@@ -182,40 +193,92 @@ def long_term_total_payoff(opening_payoffs, exp_p, delta):
...
+    player1_moves: list of moves of player1
+    player2_mem1_strategy: (init, pcc, pcd, pdc, pdd)  (tuple of Actions)
+    """
+    last = (player1_moves[0], player2_mem1_strategy[0])
+    history = []
+    history.append(last)
+    for i in range(1,len(player1_moves)):
```
Comment (Nikoleta-v3) [inline src/mockup.py]: This whole loop can be:

```python
for i, m1 in enumerate(player1_moves)
```

File: src/mockup.py [hunk 4]
```diff
@@ -182,40 +193,92 @@ def long_term_total_payoff(opening_payoffs, exp_p, delta):
...
+            m2 = player2_mem1_strategy[2]
+        elif last == (D,D):
+            m2 = player2_mem1_strategy[4]
+        else:
+            raise ValueError("Invalid history")
+        history.append((m1, m2))
+        last = (m1, m2)
+    return history
+
+def calc_payoff_from_history(history, benefit, cost):
```
Comment (Nikoleta-v3) [inline src/mockup.py]: Should we make this function more generic? Should it take R, S, T, P as arguments?

**Comment by yohm** [comment]
Thank you for the feedback. I did refactoring on the code.

**Comment by Nikoleta-v3** [comment]
Thank you @yohm! I just realised that `for i,m1 in enumerate(player1_moves[1:])` can just be `for m1 in player1_moves[1:]:` since `I` is never used.

Otherwise everything looks good 👍🏻

**Comment by yohm** [comment]
Thank you for catching it. I polished the code.

**Pull request merged**